### PR TITLE
coproduct of categories

### PR DESCRIPTION
**Pull request by maxsnew**

### CODE REVIEW

File: Cubical/Categories/Constructions/Coproduct.agda [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    cong lift (C .⋆Assoc (f .lower) (g .lower) (h .lower))
+  _⊎_ .⋆Assoc {inr _} {inr _} {inr _} {inr _} f g h =
+    cong lift (D .⋆Assoc (f .lower) (g .lower) (h .lower))
+  _⊎_ .isSetHom {inl _} {inl _} = isOfHLevelLift 2 (C .isSetHom)
+  _⊎_ .isSetHom {inr _} {inr _} = isOfHLevelLift 2 (D .isSetHom)
+  _⊎_ .isSetHom {inl _} {inr _} = isProp→isSet isProp⊥*
+  _⊎_ .isSetHom {inr _} {inl _} = isProp→isSet isProp⊥*
+
+module _ {C : Category ℓC ℓC'}{D : Category ℓD ℓD'} where
+  -- TODO: should these be inl and inr?
```
Comment (hejohns) [inline Cubical/Categories/Constructions/Coproduct.agda:70]: Why would it not be?

File: Cubical/Categories/Constructions/Coproduct.agda [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  Inr .F-ob = inr
+  Inr .F-hom = lift
+  Inr .F-id = refl
+  Inr .F-seq _ _ = refl
+
+  module _ {Cᴰ : Categoryᴰ (C ⊎ D) ℓCᴰ ℓDᴰ}
+           (inl-case : Section Inl Cᴰ)
+           (inr-case : Section Inr Cᴰ)
+         where
+    elim : GlobalSection Cᴰ
```
Comment (hejohns) [inline Cubical/Categories/Constructions/Coproduct.agda:87]: Ah so gluing together two local sections, over each disjunct.

File: Cubical/Categories/Constructions/Coproduct.agda [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+         where
+    elimLocal : Section F Cᴰ
+    elimLocal = GlobalSectionReindex→Section _ _
+      (elim (Reindex.introS _ inl-case) (Reindex.introS _ inr-case))
+
+  module _ {E : Category ℓE ℓE'}
+           (inl-case : Functor C E)
+           (inr-case : Functor D E)
+         where
+    rec : Functor (C ⊎ D) E
```
Comment (hejohns) [inline Cubical/Categories/Constructions/Coproduct.agda:113]: So if Cᴰ is trivially displayed (E over E), we get the classical universal property

File: Cubical/Categories/Constructions/Coproduct.agda (hunk 1 above)
Comment (maxsnew) [reply Cubical/Categories/Constructions/Coproduct.agda:70]: I meant should they be capitalized or not

**Pull request merged**